apps/emqx_connector/src/emqx_connector_redis.erl

```diff
@@ -220,6 +220,8 @@ do_query(InstId, Query, #{pool_name := PoolName, type := Type} = State) ->
             case is_unrecoverable_error(Reason) of
                 true ->
                     {error, {unrecoverable_error, Reason}};
+                false when Reason =:= ecpool_empty ->
+                    {error, {recoverable_error, Reason}};
                 false ->
                     Result
```

Comment:
Hmm, strange that it wasn't marked recoverable. It will be [considered unrecoverable](https://github.com/emqx/emqx/blob/a3a6480f005aadb65678e95a30356a20ff369e7b/apps/emqx_resource/src/emqx_resource_buffer_worker.erl#L1703-L1706) by buffer worker, right?

Reply:
Yes, currently it will. :speak_no_evil:

Reply:
Actually, there are some small transformations in the EE connector around this: https://github.com/emqx/emqx/blob/7055bd85644a9849de10644da45d881c8fdcace2/lib-ee/emqx_ee_connector/src/emqx_ee_connector_redis.erl#L114

Reply:
Heh. 🫠 Do you have an idea why is it on a different "connector level"?